src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsTag.cs

```diff
@@ -0,0 +1,187 @@
+                {
+                    Height = lineHeight,
+                    Orientation = Orientation.Horizontal,
+                    Children = { image, block }
+                },
+                CornerRadius = new CornerRadius(2),
+                // Highlighting lines are 2px buffer. So shift us up by one from the bottom so we feel centered between them.
+                Margin = new Thickness(10, top: 0, 0, bottom: 1),
+                Padding = new Thickness(1)
+            };
+
+            // This is used as a workaround to the moniker issues in blue theme
+            var editorBackground = (Color)_editorFormatMap.GetProperties("TextView Background")["BackgroundColor"];
+            ImageThemingUtilities.SetImageBackgroundColor(border, editorBackground);
+
+            border.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            view.ViewportWidthChanged += ViewportWidthChangedHandler;
+            view.LayoutChanged += View_LayoutChanged;
+
+            return new GraphicsResult(border, dispose:
+                () =>
+                {
+                    if (link is not null)
+                    {
+                        link.RequestNavigate -= HandleRequestNavigate;
+                    }
+
+                    view.ViewportWidthChanged -= ViewportWidthChangedHandler;
+                    view.LayoutChanged -= View_LayoutChanged;
+                });
+
+            void ViewportWidthChangedHandler(object s, EventArgs e)
+            {
+                if (Location is InlineDiagnosticsLocations.PlacedAtEndOfEditor)
+                {
+                    Canvas.SetLeft(border, view.ViewportRight - border.DesiredSize.Width);
+                }
+            }
+
+            void View_LayoutChanged(object sender, TextViewLayoutChangedEventArgs e)
+            {
+                if (Location is InlineDiagnosticsLocations.PlacedAtEndOfEditor)
+                {
+                    Canvas.SetLeft(border, view.ViewportRight - border.DesiredSize.Width);
+                }
+            }
+
+            void HandleRequestNavigate(object sender, RoutedEventArgs e)
+            {
+                var uri = link.NavigateUri;
+                _ = _navigateToLinkService.TryNavigateToLinkAsync(uri, CancellationToken.None);
+                e.Handled = true;
+            }
+        }
+
+        private ImageMoniker GetMoniker()
+            => _diagnostic.Severity switch
+            {
+                DiagnosticSeverity.Warning => KnownMonikers.StatusWarning,
+                _ => KnownMonikers.StatusError,
+            };
+
+        public static string GetClassificationId(string error)
+            => error switch
+            {
+                EditAndContinueErrorTypeDefinition.Name => "inline diagnostics - Edit and Continue",
+                PredefinedErrorTypeNames.SyntaxError => "inline diagnostics - syntax error",
+                _ => "inline diagnostics - compiler warning"
```

Comment:
not a fan.  better to check for the value you expect instead of defaulting to warning.  if something unexpected shows up, we can throw.